### ISSUE TITLE
VAE Config Files

### ISSUE THREAD

**Issue by jdipalma2** [cvlab-stonybrook/PathLDM#27]
I'm interested in fine-tuning a VAE and was wondering the config files you used for the final vq-f4 model in your paper?
I see the existing config files under the autoencoder directory, but they are missing some hyper-parameters like number of training epochs.
Please advise.

**Comment by srikarym** [comment]
The VAE used in our paper is a VQ-f4 VAE (VQGAN). The original LDM repo only supports KL VAE training, so we used the codebase of [taming-transformers](https://github.com/CompVis/taming-transformers). We fine-tuned the VAE for 50k iterations on 3 GPUs, with bs=12 per GPU.

Diffusers library now supports training a VQ-VAE: https://github.com/huggingface/diffusers/tree/main/examples/vqgan

You might want to track image quality metrics like PSNR / SSIM during training.

**Comment by jdipalma2** [comment]
What about the discriminator hyper-parameters and the codebook weight for the LPIPS loss?

**Comment by jdipalma2** [comment]
On second thought, would it be possible to share the config file you used for this phase?
I have managed to make the code run, but not sure if my results are correct.
They look strange as the total loss is increasing and I'm not sure if one of my settings is bad.

I appreciate all the help!

**Comment by srikarym** [comment]
We used the same hyperparameters as those from the [taming transformers repo](https://github.com/CompVis/taming-transformers/blob/master/configs/custom_vqgan.yaml)

You might have to adjust the `disc_start` parameter, which controls the training iteration to add discriminator loss. 

```
model:
  base_learning_rate: 4.5e-06
  target: ldm.models.autoencoder.VQModel
  params:
    embed_dim: 3
    n_embed: 8192
    monitor: val/rec_loss
    ddconfig:
      double_z: false
      z_channels: 3
      resolution: 256
      in_channels: 3
      out_ch: 3
      ch: 128
      ch_mult:
      - 1
      - 2
      - 4
      num_res_blocks: 2
      attn_resolutions: []
      dropout: 0.0
    lossconfig:
      target: taming.modules.losses.vqperceptual.VQLPIPSWithDiscriminator
      params:
        disc_conditional: False
        disc_in_channels: 3
        disc_start: 10000
        disc_weight: 0.8
        codebook_weight: 1.0
data:
  target: main.DataModuleFromConfig
  params:
    batch_size: 12
    wrap: false
    train:
      target: ldm.data.tcga_brca.TCGADataset
      params:
        config:
          root: /home/myellapragad/TCGA_dataset
          crop_size: 256
          split: train
    validation:
      target: ldm.data.tcga_brca.TCGADataset
      params:
        config:
          root: /home/myellapragad/TCGA_dataset
          crop_size: 256
          split: test
lightning:
  callbacks:
    image_logger:
      target: main.ImageLogger
      params:
        batch_frequency: 1000
        max_images: 8
        increase_log_steps: True

  trainer:
    benchmark: True
    accumulate_grad_batches: 2
 ```

**Comment by jdipalma2** [comment]
So to confirm, you used 3 GPUs with batch size 12 each, and 2 gradient accumulation steps, for an effective batch size of 72?

**Comment by srikarym** [comment]
correct

**Comment by jdipalma2** [comment]
One last question and appreciate all the help.

Did you use disc_start=10000 or experiment and find a better value?

**Comment by srikarym** [comment]
I suggest setting it to a higher value (50,000). Once you observe an improvement in reconstructions, you could introduce adversarial loss.
